compiler/src/dotty/tools/dotc/typer/ErrorReporting.scala

```diff
@@ -154,7 +157,9 @@ object ErrorReporting {
       val found1 = reported(found)
       reported.setVariance(-1)
       val expected1 = reported(expected)
-      TypeMismatch(found1, expected1, whyNoMatchStr(found, expected), postScript)
+      val (found2, expected2) =
```

Comment:
Improvement: Avoid a type error message where "found" looks like it is a subtype of "expected".